lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -198,23 +208,36 @@ def mock_run(args, cwd, stderr):
         tree.create("ABC", source_tarball)
 
         tarball = tree.find_dataset(md5)
+        assert tarball.resource_id == md5
+        assert tarball.resource_id in tree
         tree.unpack(md5)
+        assert tarball.unpacked
 
     def test_unpack_with_exception(
         self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
     ):
         """Test to check the unpacking functionality of the FileTree when
         an Exception occured"""
         source_tarball, source_md5, md5 = tarball
-
-        def mock_run(args, cwd, stderr):
+        tarball_name = source_tarball.name[0 : -len(".tar.xz")]
+
+        def mock_run(args, cwd, stdin, stderr, stdout):
+            assert args == [
+                "tar",
+                "-xf",
+                (server_config.ARCHIVE / "ABC" / source_tarball.name),
+            ]
+            assert cwd == server_config.INCOMING / "ABC" / tarball_name
+            assert stderr == subprocess.DEVNULL
```

Comment:
These assertions are OK, but I wouldn't bother with them unless there is reason to think that the behavior that they are testing is different from the behavior tested at line 198-201.  (E.g., is there a reason why you are checking `stderr` and not `stdin` nor `stdout`, here?)  (Ditto lines 262-263.)